common/src/jni/main/cpp/conscrypt/jniutil.cc

```diff
@@ -76,6 +84,10 @@ void init(JavaVM* vm, JNIEnv* env) {
             getMethodRef(env, openSslInputStreamClass, "gets", "([B)I");
     outputStream_writeMethod = getMethodRef(env, outputStreamClass, "write", "([B)V");
     outputStream_flushMethod = getMethodRef(env, outputStreamClass, "flush", "()V");
+    byteBuffer_isReadMethod = getMethodRef(env, byteBufferClass, "isReadOnly", "()Z");
```

Comment:
You can get rid of `_isReadMethod` and `_isDirectMethod` once you remove the read-only test from `evp_aead_ctx_op_buf()`.

Reply:
thanks for catching this!